hnn_core/batch_simulate.py

```diff
@@ -51,7 +56,41 @@ def __init__(self, set_params, net_name='jones', tstop=170,
             :meth:`~hnn_core.dipole.Dipole.smooth` and
             :meth:`~hnn_core.dipole.Dipole.scale` methods instead.
             Default: False.
+        save_outputs : bool, optional
+            Whether to save the simulation outputs to files. Default is False.
+        file_path : str, optional
+            The path to save the simulation outputs.
+            Default is './sim_results'.
+        batch_size : int, optional
+            The maximum number of simulations saved in a single file.
+            Default is 100.
+        overwrite : bool, optional
+            Whether to overwrite existing files and create file paths
+            if they do not exist. Default is True.
+        summary_func : callable, optional
```

Comment:
you can use `callable()` to add a check that this and `set_params` are both callables